snowflake/src/test/java/org/apache/iceberg/snowflake/SnowflakeCatalogTest.java

```diff
@@ -0,0 +1,277 @@
+  public void testListTablesWithinDB() {
+    String dbName = "DB_1";
+    Assertions.assertThat(catalog.listTables(Namespace.of(dbName)))
+        .containsExactly(
+            TableIdentifier.of("DB_1", "SCHEMA_1", "TAB_1"),
+            TableIdentifier.of("DB_1", "SCHEMA_1", "TAB_2"));
+  }
+
+  @Test
+  public void testListTablesWithinNonexistentDB() {
+    String dbName = "NONEXISTENT_DB";
+    Assertions.assertThatExceptionOfType(RuntimeException.class)
+        .isThrownBy(() -> catalog.listTables(Namespace.of(dbName)))
+        .withMessageContaining("does not exist")
+        .withMessageContaining(dbName);
+  }
+
+  @Test
+  public void testListTablesWithinSchema() {
+    String dbName = "DB_2";
+    String schemaName = "SCHEMA_2";
+    Assertions.assertThat(catalog.listTables(Namespace.of(dbName, schemaName)))
+        .containsExactly(
+            TableIdentifier.of("DB_2", "SCHEMA_2", "TAB_3"),
+            TableIdentifier.of("DB_2", "SCHEMA_2", "TAB_4"));
+  }
+
+  @Test
+  public void testListTablesWithinNonexistentSchema() {
+    String dbName = "DB_2";
+    String schemaName = "NONEXISTENT_SCHEMA";
+    Assertions.assertThatExceptionOfType(RuntimeException.class)
+        .isThrownBy(() -> catalog.listTables(Namespace.of(dbName, schemaName)))
+        .withMessageContaining("does not exist")
+        .withMessageContaining("DB_2.NONEXISTENT_SCHEMA");
+  }
+
+  @Test
+  public void testLoadS3Table() {
+    Table table = catalog.loadTable(TableIdentifier.of(Namespace.of("DB_1", "SCHEMA_1"), "TAB_1"));
+    Assertions.assertThat(table.location()).isEqualTo("s3://tab1/");
+  }
+
+  @Test
+  public void testLoadAzureTable() {
+    Table table = catalog.loadTable(TableIdentifier.of(Namespace.of("DB_2", "SCHEMA_2"), "TAB_3"));
+    Assertions.assertThat(table.location())
+        .isEqualTo("wasbs://mycontainer@myaccount.blob.core.windows.net/tab1/");
+  }
+
+  @Test
+  public void testLoadGcsTable() {
+    Table table = catalog.loadTable(TableIdentifier.of(Namespace.of("DB_3", "SCHEMA_3"), "TAB_5"));
+    Assertions.assertThat(table.location()).isEqualTo("gs://tab5/");
+  }
+
+  @Test
+  public void testLoadTableWithMalformedTableIdentifier() {
+    Assertions.assertThatExceptionOfType(IllegalArgumentException.class)
+        .isThrownBy(
+            () ->
+                catalog.loadTable(
+                    TableIdentifier.of(Namespace.of("DB_1", "SCHEMA_1", "BAD_NS_LEVEL"), "TAB_1")))
+        .withMessageContaining("level")
+        .withMessageContaining("DB_1.SCHEMA_1.BAD_NS_LEVEL");
+    Assertions.assertThatExceptionOfType(IllegalArgumentException.class)
+        .isThrownBy(
+            () -> catalog.loadTable(TableIdentifier.of(Namespace.of("DB_WITHOUT_SCHEMA"), "TAB_1")))
+        .withMessageContaining("level")
+        .withMessageContaining("DB_WITHOUT_SCHEMA.TAB_1");
+  }
+
+  @Test
+  public void testCloseBeforeInitialize() throws IOException {
+    catalog = new SnowflakeCatalog();
+    catalog.close();
```

Comment:
what does this test exactly?

Reply:
Testing that it's safe to call close() before initialize() without unexpected exceptions; the thought is that a caller might decide to abort using a particular catalog for whatever reason without initializing them, and might have some auto-close helpers, for example.

Added a comment to explain here.

Reply:
I think it would still be good to add some assertion here as this looks weird without any assertion

Reply:
Good point, added assertion that despite this no-exception close() call, we didn't close the injected `fakeClient`.